# HG changeset patch
# User Bart van Strien <bart.bes+projects@gmail.com>
# Date 1526808449 -7200
# Node ID bada7dbb884ad74ec5a964ec769239a8fb8a3cb9
# Parent  76130a90751a561a3bfc9f280e5a2e6ccfba6034
Always link against pthread when using autotools (resolves #1410)

Theoretically I could enable it on a case-by-case basis, but as of now glslang,
physfs and the thread module all require pthread, so I might as well just
always link it in.

diff --git a/platform/unix/configure.ac b/platform/unix/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -63,6 +63,7 @@
 ACLOVE_DEP_SDL2
 ACLOVE_DEP_LIBM
 ACLOVE_DEP_ZLIB
+ACLOVE_DEP_PTHREAD
 
 # Conditional dependencies
 AS_VAR_IF([enable_module_audio], [yes], [ACLOVE_DEP_OPENAL], [])
diff --git a/platform/unix/deps.m4 b/platform/unix/deps.m4
--- a/platform/unix/deps.m4
+++ b/platform/unix/deps.m4
@@ -22,6 +22,9 @@
 AC_DEFUN([ACLOVE_DEP_SDL2], [
 	AM_PATH_SDL2([], [], [LOVE_MSG_ERROR([SDL 2])])])
 
+AC_DEFUN([ACLOVE_DEP_PTHREAD], [
+	AC_SEARCH_LIBS([pthread_create], [pthread], [], [LOVE_MSG_ERROR([the POSIX threads library])])])
+
 # does not use pkg-config because of the FILE_OFFSET_BITS.. bit
 AC_DEFUN([ACLOVE_DEP_MPG123], [
 	AC_SEARCH_LIBS([mpg123_open_feed], [mpg123], [],
